### PR TITLE
add keyboard: pierce

### PR DESCRIPTION
**Pull request by durken1**
## Description

add new keyboard: pierce, including two keymaps (default + durken1)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by durken1** [comment]
I minimized the default keymap and double checked the PR checklist. I would appreciate a comment if there is something else I need to adjust.

**Comment by durken1** [comment]
I have gone through the PR Checklist again and made adjustments. The one thing I am unsure about is using QMK abstractions for I2C. I use the pro micro running QMK as slave but can only find drivers for running in master mode. Would be nice if someone could point me in the right direction

File: keyboards/pierce/matrix.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#include "i2c_slave.h"
```
Comment (drashna) [inline keyboards/pierce/matrix.c:18]: i2c_master should probably be used here, not i2c_slave.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/pierce/rules.mk [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+CUSTOM_MATRIX = lite
+
+# project specific files
+SRC += matrix.c \
+       i2c_slave.c
```
[the remark this answers is not in the document]
Comment (durken1) [reply keyboards/pierce/rules.mk]: Yes, it's using a wireless MCU (master) that sends keypresses to the QMK device (slave)

File: keyboards/pierce/matrix.c (hunk 1 above)
Comment (durken1) [reply keyboards/pierce/matrix.c:18]: See coment above :)

**Comment by durken1** [comment]
I have made the requested changes except the ones regarding i2c slave mode, since the MCU running QMK is operating as slave and a separate MCU as master.

File: keyboards/pierce/keymaps/durken1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+ifeq ($(strip $(TRACKPOINT)), yes)
+  PS2_MOUSE_ENABLE = yes
+  PS2_USE_USART = yes
+  OPT_DEFS += -DUSART
+endif
+
+ifeq ($(strip $(AUTO_BUTTONS)), yes)
+  OPT_DEFS += -DAUTO_BUTTONS
+endif
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/rules.mk:14]: These should be moved to `post_rules.mk`

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/pierce/rules.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/pierce/rules.mk]: Is the wireless mcu sending the HID information to the host, or is the QMK device?

[the remark this answers is not in the document]
Comment (durken1) [reply keyboards/pierce/rules.mk]: The qmk device. The wireless MCU is either receiving keypresses wirelessly or reading keypresses directly depending on configuration and sending them to the qmk device.

File: keyboards/pierce/pierce.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	K20, K21, K22, K23, K24,	K25, K26, K27, K28, K29, \
+	            K32, K33, K34,    K35, K36, K37  \
+) { \
+	{ K00,  K01,  K02,  K03,  K04,  K05,  K06,  K07,  K08,  K09 }, \
+	{ K10,  K11,  K12,  K13,  K14,  K15,  K16,  K17,  K18,  K19 }, \
+	{ K20,  K21,  K22,  K23,  K24,  K25,  K26,  K27,  K28,  K29 }, \
+	{ KC_NO, KC_NO, K32,  K33,  K34,  K35,  K36,  K37,  KC_NO, KC_NO }  \
+}
+
+#define LAYOUT LAYOUT_split_3x5_3
```
Comment (drashna) [inline keyboards/pierce/pierce.h:33]:
```suggestion
```

File: keyboards/pierce/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+#include QMK_KEYBOARD_H
+
+enum layers { 
+    BASE, 
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [BASE] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/default/keymap.c:24]:
```suggestion
  [BASE] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/keymaps/durken1/keymap.c [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+const uint16_t PROGMEM odia_combo[] = {SE_U, SE_K, COMBO_END};
+
+combo_t key_combos[COMBO_COUNT] = {
+	[WF_ARNG] = COMBO(arng_combo, SE_ARNG),
+	[EI_ADIA] = COMBO(adia_combo, SE_ADIA),
+	[UK_ODIA] = COMBO(odia_combo, SE_ODIA)
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [BASE] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/keymap.c:93]:
```suggestion
  [BASE] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/keymaps/durken1/keymap.c [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+         SE_Z,    SE_X,    SE_C,    SE_D,    SE_V,                         SE_J,    SE_H, SE_COMM,  SE_DOT, SE_MINS,
+  //|--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------|
+                                  KC_ESC, NUM_SPC,  KC_TAB,     FN_ENT, SYM_BSP,  KC_DEL
+                             //`--------------------------'  '--------------------------'
+
+  ),
+
+
+  [MBO] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/keymap.c:108]:
```suggestion
  [MBO] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/keymaps/durken1/keymap.c [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+      _______, _______, _______, _______, _______,                      KC_BTN3, _______, _______, _______, _______,
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+      _______, _______, _______, _______, _______,                      _______, _______, _______, _______, _______,
+  //|--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------|
+                                 _______, _______, _______,    _______, KC_BSPC, _______
+                             //`--------------------------'  '--------------------------'
+
+  ),
+
+  [SYM] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/keymap.c:122]:
```suggestion
  [SYM] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/keymaps/durken1/keymap.c [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+      SE_EXLM,  SE_DLR, SE_LPRN, SE_LBRC, _______,                      _______, SE_RBRC, SE_RPRN, SE_HASH, SE_QUES,
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+      _______, SE_PERC, SE_LCBR, SE_LABK, _______,                      _______, SE_RABK, SE_RCBR, SE_AMPR, _______,
+  //|--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------|
+                                   SE_AT,  SE_EQL, SE_DQUO,    _______, _______, _______
+                             //`--------------------------'  '--------------------------'
+  ),
+
+  [NUM] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/keymap.c:135]:
```suggestion
  [NUM] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/keymaps/durken1/keymap.c [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+      SE_PLUS, SE_MINS, SE_ASTR, SE_SLSH, SE_CIRC,                      _______,    SE_4,    SE_5,    SE_6, _______,
+  //|--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------|
+      _______, _______, KC_VOLD, KC_VOLU, _______,                       KC_SPC,    SE_1,    SE_2,    SE_3, _______,
+  //|--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------|
+                                 _______, _______, _______,    SE_COMM,   SE_0,  SE_DOT
+                             //`--------------------------'  '--------------------------'
+  ),
+
+  [FN] = LAYOUT(
```
Comment (drashna) [inline keyboards/pierce/keymaps/durken1/keymap.c:148]:
```suggestion
  [FN] = LAYOUT_split_3x5_3(
```

File: keyboards/pierce/rules.mk [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+CUSTOM_MATRIX = lite
+SRC += matrix.c
+QUANTUM_LIB_SRC += i2c_slave.c
```
Comment (drashna) [inline keyboards/pierce/rules.mk:22]:
```suggestion
QUANTUM_LIB_SRC += i2c_slave.c

LAYOUTS = split_3x5_3
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/pierce/pierce.h (hunk 4 above)
Comment (durken1) [reply keyboards/pierce/pierce.h]: Thanks! and fixed :)

**Comment by durken1** [comment]
@drashna Sorry for the ping, but I have made all the requested changes. Are there any more questions regarding the i2c or could I get a review?

File: keyboards/pierce/keymaps/default/config.h [hunk 12]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2022 durken (https://github.com/durken1/)
```
Comment (fauxpark) [inline keyboards/pierce/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/pierce/keymaps/default/rules.mk [hunk 13]
```diff
@@ -0,0 +1,3 @@
+# Change to yes to enable feature
```
Comment (fauxpark) [inline keyboards/pierce/keymaps/default/rules.mk:1]:
```suggestion
```

File: keyboards/pierce/keymaps/durken1/rules.mk [hunk 14]
```diff
@@ -0,0 +1,5 @@
+# Change to yes to enable feature
+EXTRAKEY_ENABLE = yes # Audio control and System control
```
Comment (fauxpark) [inline keyboards/pierce/keymaps/durken1/rules.mk:2]:
```suggestion
EXTRAKEY_ENABLE = yes
```

File: keyboards/pierce/readme.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
+# Pierce
+
+![Pierce](https://i.imgur.com/JOUH1UI.jpeg)
+*A modular wireless / semi-wireless split 36 key keyboard. Wireless version uses a dongle as a receiver for both halves while the semi-wireless version uses the wired half as a receiver for the wireless half.
+Supports MX, Choc V1 and Alps switches.*
```
Comment (fauxpark) [inline keyboards/pierce/readme.md:5]:
```suggestion
![Pierce](https://i.imgur.com/JOUH1UIh.jpeg)

A modular wireless / semi-wireless split 36 key keyboard. Wireless version uses a dongle as a receiver for both halves while the semi-wireless version uses the wired half as a receiver for the wireless half.
Supports MX, Choc V1 and Alps switches.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**